opentelemetry-sdk/src/metrics/meter.rs

```diff
@@ -25,7 +25,7 @@ use crate::metrics::{
 const INSTRUMENT_NAME_MAX_LENGTH: usize = 255;
 // maximum length of instrument unit name
 const INSTRUMENT_UNIT_NAME_MAX_LENGTH: usize = 63;
-const INSTRUMENT_NAME_ALLOWED_NON_ALPHANUMERIC_CHARS: [char; 4] = ['-', '.', '-', '/'];
+const INSTRUMENT_NAME_ALLOWED_NON_ALPHANUMERIC_CHARS: [char; 4] = ['_', '.', '-', '/'];
```

Comment:
https://github.com/open-telemetry/opentelemetry-rust/issues/1265 This has some unit-tests added, wondering why the unittests did not caught this?

Reply:
I think the existing unit-test doesn't catch this. Good to add one.

Reply:
+1 for unit tests

Reply:
https://github.com/open-telemetry/opentelemetry-rust/issues/1219 is opened to track adding tests. Resolving this conversation and would suggest to merge this PR with the fix.